### ISSUE TITLE
Problems with RNN

### ISSUE THREAD

**Issue by giovannivolpe** [DeepTrackAI/deeplay#49]
There seem to be some problems with the RNN:
- [ ] during training the log messages appear empty as "...."
- [ ] the dropout as parameter seems not to be working correctly

See the example rnn.py in https://github.com/DeepTrackAI/DeepLearningCrashCourse/pull/25

**Comment by Henrik-KM** [comment]
Cannot recreate the empty log messages on my end.

The dropout though is a known issue, and is due to how pyTorch overrides dropout to always equal zero if n_layers=1. It's solved by having a RecurrentBlock with a custom dropout layer, but if the user sets the dropout parameter of the internal layers directly then it will fail (i.e. automatically overwritten by torch). We need a long-term solution that prevents users from changing the wrong parameter.

**Comment by cmanzo** [comment]
`rnn = dl.rnn(
    in_features,
    hidden_features=[100],
    out_features=model_number,
    rnn_type="RNN",
    dropout=0.2, 
)`
seems to properly set the droput without needing .configure afterwards.

I'm not sure if it's the same thing that @giovannivolpe was mentioning regarding the training log: during training with validation, the validation slider continuously appears and at each epoch leaves a white box that it's quite annoying. I guess this is a general feature though.

**Comment by Henrik-KM** [comment]
@cmanzo yes, that should work since dropout is defined in dl.rnn through the custom dropout layer. Though setting e.g.

```
rnn = dl.rnn( in_features, hidden_features=[100], out_features=model_number, rnn_type="RNN", dropout=0 )
rnn.blocks.layer[0].configure(dropout=0.2)
```

will not work, dropout will still be 0. Rather, you would need to do this:

```
rnn = dl.rnn( in_features, hidden_features=[100], out_features=model_number, rnn_type="RNN", dropout=0 )
rnn.blocks.dropout.configure(0.2)
```

**Comment by cmanzo** [comment]
as it is now, it seems fine to me and consistent with deeplay. The main issue I was having yesterday was that setting `dropout=0.2` within `dl.rnn` was not working and has been solved.

**Comment by BenjaminMidtvedt** [comment]
@cmanzo @Henrik-KM can this be closed?

**Comment by cmanzo** [comment]
> @cmanzo @Henrik-KM can this be closed?

yes!
